### ISSUE TITLE
[expo-module-template] Missing expo-modules-core as dependencies

### ISSUE THREAD

**Issue by giautm** [expo/expo#15878]
### Summary

Run `npx create-expo-module hello-world` cause a error when running `build` commands
```sh
📦 Installing dependencies...
 🛠  Compiling TypeScript files...
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

Error: yarn exited with non-zero code: 2
    at ChildProcess.completionListener (/Users/giautm/.npm/_npx/62900f925a4b91dc/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:510:26)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
    ...
    at spawnAsync (/Users/giautm/.npm/_npx/62900f925a4b91dc/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at run (/Users/giautm/.npm/_npx/62900f925a4b91dc/node_modules/create-expo-module/build/create-expo-module.js:137:41)
    at postActionsAsync (/Users/giautm/.npm/_npx/62900f925a4b91dc/node_modules/create-expo-module/build/create-expo-module.js:145:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Command.main (/Users/giautm/.npm/_npx/62900f925a4b91dc/node_modules/create-expo-module/build/create-expo-module.js:61:5) {
  pid: 19433,
  output: [ '', '' ],
  stdout: '',
  stderr: '',
  status: 2,
  signal: null
}
```

The detail error:
```
yarn run build
yarn run v1.22.15
$ tsc
src/ExpoSdk.ts:1:36 - error TS2307: Cannot find module 'expo-modules-core' or its corresponding type declarations.

1 import { NativeModulesProxy } from 'expo-modules-core';
                                     ~~~~~~~~~~~~~~~~~~~

src/ExpoSdkView.tsx:1:42 - error TS2307: Cannot find module 'expo-modules-core' or its corresponding type declarations.

1 import { requireNativeViewManager } from 'expo-modules-core';
                                           ~~~~~~~~~~~~~~~~~~~


Found 2 errors.

error Command failed with exit code 2.
```

### Reproducible demo

Run `npx create-expo-module hello-world` in a clean directory

**Comment by brentvatne** [comment]
we should add this as a dev dependency cc @tsapeta

**Comment by giautm** [comment]
🆘 ét-o-ét, not stale

**Comment by gtokman** [comment]
@brentvatne Was there a fix for this in PR #16945?
